test: update integration tests and add Nexus client functionality

kycheng opened the pull request:
- Added environment variables for updating files in build-nexus-test-image.yaml.
- Modified test image in integration-test.yaml.
- Updated Dockerfile to use Python base image and install necessary dependencies.
- Added Nexus client implementation to support creating and managing Nexus repositories.
- Added new integration test cases for Maven, NPM, and PyPI.
- Updated feature files to support new Nexus operations.

kycheng commented:
/nexus-test

airycanon commented on testing/features/network.feature:
```diff
@@ -28,6 +28,12 @@
       """
+    并且 执行 "添加本地域名解析" 脚本成功
+      | command |
+      | ./hack/add-host.sh test-ingress-http.example.com <ingress-ip> |
```
多个工具并发执行的时候可能，ingress 可能会冲突。

airycanon commented on testing/features/network.feature:
```diff
@@ -55,6 +61,12 @@
       """
+    并且 执行 "添加本地域名解析" 脚本成功
+      | command |
+      | ./hack/add-host.sh test-ingress-https.example.com <ingress-ip> |
```
同上

airycanon commented on testing/steps/nexus.go:
```diff
@@ -0,0 +1,63 @@
+	for _, check := range checks {
+		_, err := resource.AssertResource(ctx, check)
+		if err != nil {
+			log.Error("check Sonarqube instance condition failed", zap.Error(err), zap.String("name", check.Name))
```
Sonarqube => Nexus

kycheng committed:
test: update integration tests and add Nexus client functionality

- Added environment variables for updating files in build-nexus-test-image.yaml.
- Modified test image in integration-test.yaml.
- Updated Dockerfile to use Python base image and install necessary dependencies.
- Added Nexus client implementation to support creating and managing Nexus repositories.
- Added new integration test cases for Maven, NPM, and PyPI.
- Updated feature files to support new Nexus operations.
